src/atomate2/vasp/sets/core.py

```diff
@@ -543,6 +547,135 @@ def incar_updates(self) -> dict:
+        Choose a higher number if you are doing an LOPTICS calculation.
+    **kwargs
+        Other keyword arguments that will be passed to :obj:`VaspInputGenerator`.
+    """
+
+    reciprocal_density: float = 100
+    mode: str = "STATIC"
+    copy_wavecar: bool = True
+    nbands_factor: int = 5
+    ncores: int = 16
+    nbands: int | None = None
+    force_gamma: bool = True
+    inherit_incar: bool = True  # inherit incar from previous run if available
+    SUPPORTED_MODES = ("DIAG", "GW", "STATIC")
+    config_dict: dict = field(default_factory=lambda: _BASE_MVLGW_VASP_SET)
+
+    # CONFIG = _load_yaml_config("MVLGWSet")
+
+    def __post_init__(self) -> None:
+        """Validate input settings."""
+        super().__post_init__()
+        self.mode = mode = self.mode.upper()
+
+        if mode not in MVLGWSetGenerator.SUPPORTED_MODES:
+            raise ValueError(
+                f"Invalid {mode=}, supported modes are "
+                f"{', '.join(map(repr, MVLGWSetGenerator.SUPPORTED_MODES))}"
+            )
+
+    @property
+    def kpoints_updates(
+        self,
+    ) -> dict:
+        """Get updates to the kpoints configuration for this calculation type."""
+        # Generate gamma center k-points mesh grid for GW calc, which is requested
+        # by GW calculation.
+        return {"reciprocal_density": self.reciprocal_density}
+
+    @property
+    def incar_updates(
+        self,
+    ) -> dict:
+        """Get updates to the INCAR config for this calculation type."""
+        updates: dict[str, str | float | int | bool] = {}
+        nbands = (
+            int(self.prev_vasprun.parameters["NBANDS"])
+            if self.prev_vasprun is not None
+            else None
+        )
+
+        if self.mode == "STATIC":
+            # TODO: the VaspInputGenerator._get_incar() method
+            #       always changes the ISMEAR and SIGMA parameters.
+            #       So, I manually set them here.
+            updates.update(
+                {
+                    "ALGO": "Normal",
+                    "ISMEAR": 0,
+                    "SIGMA": 0.01,
+                }
+            )
+        elif self.mode == "DIAG":
+            # Default parameters for diagonalization calculation.
+            updates.update(
+                {
+                    "ALGO": "Exact",
+                    # In pymatgen/io/vasp/MVLGWSet.yaml,
+                    # the default value of NELM is 100;
+                    # but in the pymatgen.io.vasp.sets.MVLGWSet,
+                    # the default value of NELM for DIAG calculation
+                    # is manually set to be 1.
+                    # If 1 were used, the NSCF calculation would not converge,
+                    # and Custodian would not proceed with later calculations.
+                    # Here, I decide to keep the default value of NELM
+                    # to be 100 as defined in pymatgen.io.vasp.sets MVLGWSet.yaml.
+                    "NELM": 100,
```

Comment:
FYI NELM should be 1 as per the VASP manual - exact diagonalisation already converges after 1 step

Reply:
Thanks for the information. I understand. However, this is due to the fact that custodian will mistakenly treat the results of NEML = 1 as an unverged calculation. Thus, currently I set NELM = 100 and enforced ICHARG=11 (constant charge) to make it converge, so that I can pass custodian checks and proceed to G0W0 calculation.

Reply:
By the way, I also notice that `NonSCFMaker` in `atomate2.vasp.jobs.core` has the same issue. Although it finishes gracefully, custodian will still complain about being not converged.

Reply:
This should be fixed by updating the handlers for custodian

Reply:
(nevermind, just saw this was handled otherwise)